internal/mesh/internal/types/proxy_configuration.go

```diff
@@ -49,3 +53,183 @@ func MutateProxyConfiguration(res *pbresource.Resource) error {
 
 	return res.Data.MarshalFrom(&proxyCfg)
 }
+
+func ValidateProxyConfiguration(res *pbresource.Resource) error {
+	decodedProxyCfg, decodeErr := resource.Decode[*pbmesh.ProxyConfiguration](res)
+	if decodeErr != nil {
+		return resource.NewErrDataParse(decodedProxyCfg.GetData(), decodeErr)
+	}
+	proxyCfg := decodedProxyCfg.GetData()
+
+	var err error
+
+	if selErr := catalog.ValidateSelector(proxyCfg.Workloads, false); selErr != nil {
+		err = multierror.Append(err, resource.ErrInvalidField{
+			Name:    "workloads",
+			Wrapped: selErr,
+		})
+	}
+
+	if proxyCfg.GetDynamicConfig() == nil && proxyCfg.GetBootstrapConfig() == nil {
+		err = multierror.Append(err, resource.ErrInvalidFields{
+			Names:   []string{"dynamic_config", "bootstrap_config"},
+			Wrapped: errMissingProxyConfigData,
+		})
+	}
+
+	// nolint:staticcheck
+	if proxyCfg.GetOpaqueConfig() != nil {
+		err = multierror.Append(err, resource.ErrInvalidField{
+			Name:    "opaque_config",
+			Wrapped: resource.ErrUnsupported,
+		})
+	}
+
+	if dynamicCfgErr := validateDynamicProxyConfiguration(proxyCfg.GetDynamicConfig()); dynamicCfgErr != nil {
+		err = multierror.Append(err, resource.ErrInvalidField{
+			Name:    "dynamic_config",
+			Wrapped: dynamicCfgErr,
+		})
+	}
+
+	return err
+}
+
+func validateDynamicProxyConfiguration(cfg *pbmesh.DynamicConfig) error {
+	if cfg == nil {
+		return nil
+	}
+
+	var err error
+
+	// Error if any of the currently unsupported fields is set.
+	if cfg.GetMutualTlsMode() != pbmesh.MutualTLSMode_MUTUAL_TLS_MODE_DEFAULT {
+		err = multierror.Append(err, resource.ErrInvalidField{
+			Name:    "mutual_tls_mode",
+			Wrapped: resource.ErrUnsupported,
+		})
+	}
+
+	if cfg.GetMeshGatewayMode() != pbmesh.MeshGatewayMode_MESH_GATEWAY_MODE_UNSPECIFIED {
+		err = multierror.Append(err, resource.ErrInvalidField{
+			Name:    "mesh_gateway_mode",
+			Wrapped: resource.ErrUnsupported,
+		})
+	}
+
+	if cfg.GetAccessLogs() != nil {
+		err = multierror.Append(err, resource.ErrInvalidField{
+			Name:    "access_logs",
+			Wrapped: resource.ErrUnsupported,
+		})
+	}
+
+	if cfg.GetEnvoyExtensions() != nil {
```

Comment:
It might also be worth removing Envoy extensions and escape hatches until we figure out where we want to put them too.

Reply:
Yeah we can definitely remove it for now.